### ISSUE TITLE
[8.0.0] Regression: mapped types no longer marked as exports (import/named)

### ISSUE THREAD

**Issue by nevir** [eslint/typescript-eslint-parser#382]
Looks like there's a regression in 8.0 (this was working on 7.0) where it's no longer considering mapped types as exported - other types are being marked as exports (aliases, interfaces, unions, etc)

This can be seen when using [import/named](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/named.md)

---

**What version of TypeScript are you using?**
2.5.2

**What version of `typescript-eslint-parser` are you using?**
8.0.0

**What code were you trying to parse?**

`file1.ts`
```ts
export type Foo = { [key: string]: any };
```

`file2.ts`
```ts
import { Foo } from './file1';

const foo: Foo = { one: 1 };
```

**What did you expect to happen?**

No errors

**What happened?**

```
Foo not found in './file1' [import/named]
```

**Comment by JamesHenry** [comment]
I can't think of what could have changed about this within `typescript-eslint-parser` between the two versions.

Have you tried TypeScript `2.5.x` and typescript-eslint-parser `7.0.0` together? Just want to rule out the chance that this is down to the move from TypeScript 2.4 -> 2.5, rather than version of typescript-eslint-parser itself.

**Comment by nevir** [comment]
Yeah :( It's currently passing with TS `2.5.2` and parser `7.0.0`.  If it helps, you should be able to reproduce it via this diff: https://github.com/convoyinc/apollo-cache-hermes/pull/121

**Comment by nevir** [comment]
Err hm, my local build has different errors than my CI builds.  Let me investigate further - I may have goofed up my envrionment

**Comment by nevir** [comment]
Yup, PEBKAC!

**Comment by JamesHenry** [comment]
Ok cool 😄
